### PR TITLE
Swap reward math onto the combat rating and reshape the reward curve (#1532)

### LINKED ISSUE #1532: Swap the power measure onto the combat rating and reshape the reward curve (enemy-authored bounty)
Second half of spike #1526 (design + rationale in `docs/spikes/1526-combat-rating-power-measure.md`): swap both power-measure consumers onto the combat rating (#1531) **and** reshape the reward curve. The two must land together — the honest measure and the curve only work as a pair (see spike Decision 4: under the old `ratio²`-band curve, an honest measure makes build optimization *lower* XP/hour).

Blocked by #1531. Constants (`k`, pie) come from the calibration report issue.

## XP reward: enemy-authored bounty, no upward premium

```
XP/kill = k × enemyRating × min(enemyRating / playerRating, 1)²
```

- The `±20%` band, the quadratic upward premium, and the `MaxExpRewardMultiplier` cap on the XP path all dissolve into this one continuous curve. `min(r,1)² ≤ 1`, so XP is bounded by `k × enemyRating`; `MaxExpPerGrant` stays as the tamper backstop.
- Properties (verified in the spike): XP/hour peaks at matched fights; optimizing against any at-or-above-matched enemy strictly raises XP/hour (same bounty, faster kills); anti-grind below matched is unchanged in spirit; sandbagging becomes structurally pointless (nothing pays more for being weak).
- `DefeatRewards` takes rated battlers (or precomputed ratings) instead of modifier lists; `DifficultyMultiplier`'s doc comment and semantics update; the `ToIntReward` overflow clamp stays.

## Proficiency accrual: max-normalization

```
XP_path = pie × pathActivity ÷ max(playerRating, enemyRating)
```

- Above your weight the claim rate is what a matched player would earn (no weakness premium); at or below, the existing treadmill is unchanged. The `4.0` clamp mirroring `MaxExpRewardMultiplier` retires — `activity ≤ enemyHP` bounds the claim naturally (decide whether to keep a generous safety clamp; record the choice).
- `ProficiencyXpPerVictory` retunes per the calibration report (the rating's numeric scale differs from an attribute-point sum).

## Threading

- **Live**: `BattleService.RecordVictory` rates the player from the frozen snapshot (`ToBattler`, same anti-tamper rationale as today) and the enemy from its **fielded** loadout (`enemy.BattleSkills` — spike Decision 5: rate the fight actually fought; the level roll already varies per-encounter XP).
- **Offline**: player rating computes once per away window (stationary snapshot); enemy rating per encounter (closed-form arithmetic, cheap). `OfflineBattleOutcome`/`BattleStats.PlayerPower`/`BattleCompletedEvent.PlayerPower` carry the rating (rename or re-document to reflect the new meaning).

## Tests

`DefeatRewardsTests` (pins the old sum semantics — rewrite around bounty + curve), `ProficiencyRewardService`/`ProficiencyXpCalculator` tests (max-normalization), offline simulator tests, and the `BattleServiceIntegrationTests` power-snapshot assertions. Battle-sim parity suites are untouched (no parity surface).

## Docs to update (rides this issue)

- `docs/game-design.md` — *Experience Rewards* section (band → bounty curve) and the *Proficiency XP* power definition/normalization.
- `docs/content-design.md` §3 — the difficulty & reward tuning guidance (band language).
- `docs/backend-attributes.md` — the signature-passive paragraph describing `GetModifiersWithSignaturePassive` feeding `SumCoreAttributes` (battler assembly now includes the passive naturally).
- `Game.DataAccess/Repositories/Admin/AdminClasses.cs` authoring-guard comment referencing `DefeatRewards.SumCoreAttributes`.

### LINKED ISSUE #1578: Consolidate enemy→Battler construction onto Enemy.ToBattler()
### Context

While building the combat-rating calibration report (#1533), I added `Enemy.ToBattler()` (`Game.Core/Enemies/Enemy.cs`) — a one-line factory wrapping `new Battler(new AttributeCollection(GetAttributeModifiers()), BattleSkills, Level)` — because the calibration report needed to build an enemy's battler from its fielded loadout.

That exact expression was already duplicated in two production call sites before this change:

- `Game.Application/Services/BattleService.cs` (~line 511)
- `Game.Core/Battle/Offline/OfflineProgressSimulator.cs` (~line 113)

plus several test fixtures (`TypedEnemyCombatTests`, `BattleSimulatorParityTests`, `BattleSimulatorTests`, `OfflineProgressSimulatorTests`).

### Ask

Migrate `BattleService.cs` and `OfflineProgressSimulator.cs` (and the test duplicates, where it's a clean win) to call the new `Enemy.ToBattler()` instead of re-deriving the same construction inline. This is a pure DRY cleanup with no behavior change — `ToBattler()` is byte-for-byte the same expression.

### Notes

- No behavior/output change expected — this is purely reuse, not a rewrite.
- Worth double-checking whether any of the test-file duplicates are asserting on the construction itself (in which case leave them, or update them to call the new helper if it doesn't weaken the test).

### PR DESCRIPTION
**Pull request by ginderjeremiah**
## Summary

Second half of spike #1526 (design in `docs/spikes/1526-combat-rating-power-measure.md`): swaps both power-measure consumers — the XP reward and the proficiency accrual denominator — from `DefeatRewards.SumCoreAttributes` onto the combat rating (#1531), and reshapes the reward curve to match (the two only work as a pair — see spike Decision 4).

- **XP reward: enemy-authored bounty.** `XP/kill = k × enemyRating × min(r, 1)²`, `r = enemyRating ÷ playerRating`. The old ±20% band, the quadratic upward premium, and the `MaxExpRewardMultiplier` cap all dissolve into this one continuous curve — XP/hour peaks at matched fights, optimizing at-or-above-matched strictly raises XP/hour, and sandbagging is structurally pointless.
- **Proficiency accrual: max-normalization.** `XP_path = pie × pathActivity ÷ max(playerRating, enemyRating)`, with the old `4.0` clamp dropped entirely (`pathActivity` is already bounded by the enemy's health pool per battle, so it doesn't need mirroring).
- **`k` and `pie` are calibrated against authored content** via the #1533 calibration report (re-ran it to get `CombatRatingXpScale = 0.3262` and `ProficiencyXpPerVictory = 29.2469`).
- **Threading**: live (`BattleService.RecordVictory`) rates the player from the frozen battle snapshot (same anti-tamper rationale as before) and the enemy from its fielded loadout. Offline (`OfflineProgressSimulator`) rates the player once per stationary away window and reuses it across every battle, recomputing only the (cheap, closed-form) enemy rating per encounter.
- **`PlayerPower` renamed to `PlayerRating`** throughout (`BattleStats`, `BattleCompletedEvent`, `OfflineBattleOutcome`, `DefeatRewards`) to match its new meaning.
- The retired `SumCoreAttributes`/`GetDifficultyMultiplier` helpers are kept, unchanged — `CombatRatingCalibrator` still uses them to model the old curve for its old-vs-new comparison columns, which has lasting value for spotting future pricing drift.
- Docs updated: `docs/game-design.md` (Experience Rewards + Proficiency XP), `docs/content-design.md` §3, `docs/backend-attributes.md`, `docs/backend.md`, and the `AdminClasses.cs` authoring-guard comment.

Also closes **#1578**: while threading the enemy rating through `BattleService.SimulateBattle` and `OfflineProgressSimulator.SimulateBattle`, migrated both inline enemy-`Battler` constructions onto `Enemy.ToBattler()` (a pure DRY cleanup, byte-for-byte the same expression) — the exact two production call sites #1578 asked for, touched anyway to source the ratings.

## Judgment call (flagged per the issue's own invitation)

The issue explicitly left open "decide whether to keep a generous safety clamp; record the choice" for the proficiency accrual. I dropped the clamp entirely, matching the literal spike formula — `pathActivity` is inherently bounded by the enemy's health pool per battle, so a separate multiplier clamp would be redundant, and the spike's own verified corner case ("a wall enemy isn't a per-hour training exploit") supports no clamp being needed.

## Test plan

- [x] `dotnet build` — whole solution builds clean
- [x] `dotnet test` — full suite passes (2907 tests, no regressions)
- [x] `dotnet format --verify-no-changes` — clean
- Rewrote `DefeatRewardsTests`, `ProficiencyXpCalculatorTests`, `ProficiencyRewardServiceTests`, `OfflineProgressSimulatorTests`/`OfflineProgressResultTests`, and updated the `BattleServiceIntegrationTests` power-snapshot assertions (renamed to rating-snapshot) plus `BattleStatisticsEventHandlerTests`/`MultiTypedDamageProficiencyTests`. Several integration-test fixtures needed rebalancing: a player-skill-vs-enemy-skill mismatch that was irrelevant under `SumCoreAttributes` (only raw attribute totals mattered) collapsed the new rating ratio toward zero and broke leveling assertions — fixed by rebalancing the fixtures' skill damage/cooldown so ratings stay reasonably matched while the player still wins reliably. Rewrote hardcoded numeric expectations to compute against the real `CombatRating`/`DefeatRewards` production path instead of hand-derived constants, since calibration constants may shift again later.

## Follow-up noted, not filed as new

`ProgressionGraphChecker.CheckEnemyAttributeConsumption` (#1529) still uses a v1 heuristic rather than `CombatRating.Marginal` for exact dead-stat detection — already tracked as its own follow-up in #1581 (filed by a prior session), so no new issue needed; only updated its stale `SumCoreAttributes` doc-comment reference here.

Closes #1532
Closes #1578

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_016guBmzkvHyVnp7rGeQvj7D)_

### CODE REVIEW

File: Game.Application.Tests/Services/MultiTypedDamageProficiencyTests.cs [hunk 1]
```diff
@@ -246,7 +246,7 @@ private async Task<ProficiencyAccrualResult> AccrueAsync(IServiceScope scope, in
             var player = await scope.ServiceProvider.GetRequiredService<IPlayerRepository>().GetPlayer(playerId);
             Assert.NotNull(player);
             var progress = await scope.ServiceProvider.GetRequiredService<IPlayerProgressRepository>().Load(player);
-            return service.AccrueAndApply(progress, stats, totalAttributes: Power, player, notify: false);
+            return service.AccrueAndApply(progress, stats, playerRating: Power, enemyRating: Power, player, notify: false);
```
Comment (ginderjeremiah) [inline Game.Application.Tests/Services/MultiTypedDamageProficiencyTests.cs:249]: **Should-fix:** the call site here was updated to the two-rating signature, but the matching test oracle `ExpectedXp` (lines 272-278) was not — it still computes the **retired** clamped formula:

```csharp
* Math.Min(activity / Power, ServerGameConstants.MaxExpRewardMultiplier)
```

Production now uses `activity / max(playerRating, enemyRating)` with **no clamp** (`ProficiencyXpCalculator.Split`). Today this is a no-op — every scenario's `activity` is ≤ 60 against `Power = 100`, so `activity/Power ≤ 0.6` never reaches the `4.0` cap and the clamped/unclamped results coincide — which is exactly why CI stays green while the oracle silently diverges from the code it's meant to mirror. If a future scenario (or a rebalanced `Power`) ever pushes `activity/Power ≥ 4`, this oracle would expect the clamped value while production returns the unclamped one, and the test would fail for the wrong reason (or mask a real regression).

Since the PR already touched this file to update the contract, please finish the update in-scope: drop the `Math.Min(..., MaxExpRewardMultiplier)` so the oracle is just `activity / Power`, and fix the stale `clamp(activity ÷ power)` wording in the comment on lines 269-270 / 273. (There are also a couple of stale "clamp"/"activity ÷ power" comments in `BattleStatisticsEventHandlerTests` around lines 268 and 380 — optional nits, harmless.)

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by ginderjeremiah — COMMENTED** [review]
## Review: Swap reward math onto the combat rating & reshape the reward curve (#1532, also #1578)

**Verdict:** Request changes — one should-fix (a test oracle still encodes the retired clamp). Everything else is clean; this is a strong PR. _(Posted as `COMMENT` because I'm the PR author and GitHub won't accept a gating verdict on my own PR — the intended verdict is request-changes on that single item.)_

I reviewed the full scope against `CLAUDE.md`, `docs/backend.md`, `docs/game-design.md`, and `docs/content-design.md`: the core reward/proficiency math, the live + offline threading, the model/event renames, the docs, and the tests.

### Should-fix
- **`Game.Application.Tests/Services/MultiTypedDamageProficiencyTests.cs:272-278`** — the `ExpectedXp` oracle still computes `pie × Math.Min(activity / Power, MaxExpRewardMultiplier)`, the retired clamped formula. Production now normalizes by `max(playerRating, enemyRating)` with **no clamp**. It passes today only because every scenario's `activity/Power ≤ 0.6` never trips the `4.0` cap, so the oracle silently diverges from the code it mirrors. The PR touched this file's contract, so this is in-scope cleanup — inline comment on line 249 has the fix.

### Nits (optional)
- A few stale `clamp` / `activity ÷ power` phrasings in `BattleStatisticsEventHandlerTests` comments (≈ lines 268, 380) that predate max-normalization. Harmless.

### Test coverage
Thorough and correctly reworked — no gaps:
- `DefeatRewardsTests` / `ProficiencyXpCalculatorTests` rewritten around the bounty curve and max-normalization, exercising ratings directly, including the `min(r,1)²` saturation, the anti-grind tail, no-clamp linearity, the `int`-range overflow clamp, and the non-positive-normalizer guard.
- `BattleStatisticsEventHandlerTests` correctly calls `loadedEnemy.SelectBattleSkills()` so the enemy rates its fielded loadout (mirroring live `SimulateBattle`), and the offline==live parity test rates the enemy through the real `CombatRating.Rate` so both paths share one normalizer.
- Integration expectations compute against the real `CombatRating`/`DefeatRewards` path rather than hardcoded constants — the right call given the calibration constants may shift again. `ProficiencyRewardServiceTests` recalibrates `baseXp` off the pie magnitude so it's robust to that.
- No frontend/backend battle-parity surface is touched (the rating is server-only, computed off the tick loop), consistent with the docs.

### What's good
- Clean layering: the bounty/normalization math stays in `Game.Core` (`DefeatRewards`, `ProficiencyXpCalculator`), the application layer only threads ratings and orchestrates, and the `CombatRating.Rate` `Epsilon` floor makes the strictly-positive guarantee real — so the "no non-positive-rating guard needed" claim in `DefeatRewards` holds up.
- Live vs. offline stay genuinely identical: the enemy is re-rated from its fielded loadout on both paths, and the offline path rates the stationary player once per away window — the right cost trade-off, well documented.
- The retired `SumCoreAttributes` / `GetDifficultyMultiplier` / `GetModifiersWithSignaturePassive` are kept only because the `#1533` calibrator still consumes them (verified) — not dead code — and each carries a clear "retired, kept for X" doc-comment.
- Docs across `game-design.md`, `content-design.md`, `backend.md`, and `backend-attributes.md` are updated consistently with the new model, and the `#1578` DRY consolidation onto `Enemy.ToBattler()` is folded in cleanly at exactly the two call sites it asked for.

Once the `ExpectedXp` oracle is aligned with the production formula, this is an approve. I'll hold off merging until that lands and CI is green (it's still running as of this review).

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by ginderjeremiah** [comment]
Fixed in 9dabf2f: `MultiTypedDamageProficiencyTests.ExpectedXp` now computes `pie × activity ÷ power` (no clamp), matching production's unclamped max-normalization, and updated the stale `clamp`/`activity ÷ power` comment wording there and in `BattleStatisticsEventHandlerTests`. Full suite still green (2907 tests) and `dotnet format --verify-no-changes` clean.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by ginderjeremiah** [comment]
## Claude review — #1532 consumer swap + curve reshape

**Verdict: recommend CLOSE in favor of #1603.** This PR competes directly with **#1603** (a duplicate implementation of the same issue #1532 from a parallel session). Both are correct, both are green (all-checks-passed), and both produce identical reward/proficiency math with the same calibrated constants and formulas. This is a strong implementation — the recommendation to close is a narrow architectural call, not a quality knock.

### Strengths (genuinely good, some ahead of #1603)
- **Cleaner reward unit tests.** The value-based `DefeatRewards(double playerRating, double enemyRating)` ctor lets `DefeatRewardsTests` inject ratings directly and assert against literal hand-derived numbers (`floor(K × 100)` for a 100v100 fight, `0.01` multiplier at r=0.1, etc.) — a stronger independent oracle than #1603's formula-mirroring helper.
- **More complete doc/comment hygiene.** You fixed the stale `ProgressionGraphChecker.CheckEnemyAttributeConsumption` comment (the `SumCoreAttributes` → rated-threat reference #1603 left factually wrong) and refreshed the `docs/backend.md` "Proficiency XP accrual" section, which #1603 missed.
- **Explicitly `Closes #1578`** with both production `ToBattler()` sites migrated — clean bookkeeping.
- Thorough rename/re-doc, well-flagged clamp-drop judgment call, and the noted #1581 follow-up for the `CombatRating.Marginal` migration.

### Why #1603 edges it out
- **Enemy rating is computed twice per victory.** `RecordVictory` rates the enemy for the XP reward, then `BattleStatisticsEventHandler` recomputes `CombatRating.Rate(domainEvent.Enemy.ToBattler())` for the proficiency denominator — and the offline path does the same (simulator + `OfflineProgressService`). That's a DRY violation (CLAUDE.md) and two sources of truth for one deterministic quantity. #1603 computes `EnemyRating` once in `DefeatRewards` and carries it on `BattleCompletedEvent`/`OfflineBattleOutcome` as data — which is exactly what #1532 specified ("the events carry the rating").
- **Domain isolation.** Recomputing the rating in the app-layer handler/offline service leaks reward-measurement out of the domain; backend.md frames reward setup as a domain decision, not orchestration. #1603 keeps it in `DefeatRewards` and threads only scalars up.
- **Constant precision.** `ProficiencyXpPerVictory = 29.2469` (raw) drifts a fraction of a thousandth against the persisted `numeric(18,3)` XP scale; #1603 rounds to 3 decimals so a full-pie claim round-trips through storage exactly.

### If this were the one to merge instead
The must-fix would be collapsing the duplicate enemy-rating computation (expose `EnemyRating` on `DefeatRewards` and carry it on the event, as #1603 does) — that's the structural gap. Everything else here is merge-ready.

Net: the two architectural advantages of #1603 (single-source enemy rating carried as event data; domain-isolated measurement) are structural and align with the project's most-emphasized principles, whereas this PR's edges (literal-value reward tests, doc hygiene, #1578 auto-close) are each cheap to port onto #1603 as follow-ups. Recommend closing this in favor of #1603.

_Reviewed by Claude._

---
_Generated by [Claude Code](https://claude.ai/code)_
